core/templates/services/i18n-language-code.service.ts

```diff
@@ -148,12 +148,12 @@ export class I18nLanguageCodeService {
   // TODO(#14645): Remove this method when translation service is extended.
   /**
    * Checks if the translation key is valid by checking if it is present
-   * in the constants file which indicates it has atleast the translation
-   * key added in en.json file.
-   * @param {string} translationKey - Translation key that needs to be
-   * checked whether is has translation in language json files.
-   * @returns {boolean} - True if translation key is present in language json
-   * files else False.
+   * in the constants file which indicates that the translation key is
```

Comment:
Is there a frontend/backend test to verify that every key in the constants.ts file is contained in en.json/qqq.json? If so, please link to it in a reply to this comment thread -- this will help ensure that developers don't subsequently break this (since the test would fail on their PR if so).

Reply:
@seanlip we don't have any such test currently, but I think it should be there. Will look into it.

Reply:
Added backend test to check if all hacky translation keys present in constants.ts are also present in en.json in `core/controllers/base_test.py` named `test_hacky_tranlsation_keys_match_constants_en`, as for qqq.json, they are verified in the test which checks if qqq.json has all keys of en.json